### PR TITLE
#161840270 Integrate date range filter

### PR DESCRIPTION
**Pull request by DennisWanjiru**
#### What does this PR do?
Integrate date-range-picker calender

#### Description of Task to be completed?
- Integrate  date-range-picker calender
- Match displayed data with the corresponding titles
- Match the current location with the displayed data
- Implement download csv actions

#### How should this be manually tested?
- Clone the repo `https://github.com/andela/travel_tool_front.git`
- Checkout to the branch:
```ft-date-range-filter-161840270```
- Install dependencies:
```yarn install```
`yarn test`
- All the tests should be passing
- Start the application:
```yarn start```

- Login to the application
- Navigate to the dashboard page on the sidebar
- You should be able to get results as shown in the screenshot below.

#### Any background context you want to provide?
1. To use the Dashboard - Create your standalone component and plug it into the dashboard component.
2. You should have the user role of a Travel Administrator or Super Administrator to be able to see the `manage` page. To do that:
  - Download `Postico` and connect to your travela database with relevant details: Db name, password, port, user
  - Click `userRoles` table and set the `roleId` to that of a Travel Administrator(29187) or Super[]Administrator(10948) 

#### What are the relevant pivotal tracker stories?
[#161840270](https://www.pivotaltracker.com/story/show/161840270)

#### Screenshots (if appropriate)
screen shot 2018-11-09 at 16 49 04

### CODE REVIEW

File: src/components/Forms/NewAccommodationForm/NewAccommodation.jsx [hunk 1]
```diff
@@ -200,6 +200,10 @@ class NewAccommodation extends PureComponent {
     try {
       let imageUrl;
       delete axios.defaults.headers.common['Authorization'];
+      console.log('====================================');
```
Comment (ikenjoku) [inline src/components/Forms/NewAccommodationForm/NewAccommodation.jsx]: Nice work @DennisWanjiru , removing these would also fix your code climate issue

File: src/views/Dashboard/DashboardContext/__tests__/FilterContext.test.js [hunk 2]
```diff
@@ -11,38 +11,38 @@ describe('<FilterContext />', () => {
...
   });
 
-  it('should update state with today\'s filter', () => {
-    wrapper.instance().handleFilter('Today');
-    const date = moment().format('YYYY-MM-DD');
-    expect(wrapper.state().filter).toBe('Today');
-    expect(wrapper.state().range.start).toBe(date);
-    expect(wrapper.state().range.end).toBe(date);
-  });
+  // it('should update state with today\'s filter', () => {
```
Comment (ikenjoku) [inline src/views/Dashboard/DashboardContext/__tests__/FilterContext.test.js]: you could remove unused code.

**Comment by DennisWanjiru** [comment]
@ikenjoku Implemented the feedback and resolved merge conflicts.

**Comment by DennisWanjiru** [comment]
@ssewilliam which comments are you talking about. Kindly clarify.

File: src/views/Analytics/index.jsx [hunk 3]
```diff
@@ -10,49 +10,60 @@ import flightLand from '../../images/icons/flight_land.svg';
...
-          stats={people_visiting}
-          title={`No. of People visiting ${this.city} Center`}
-          icon={flightTakeoff}
-          color="brown-orange"
-        />
-        <AnalyticsCard title="Average Travel RequestLead Time" data={analytics.success && travel_lead_time_breakdown.lead_times} chart />
+        { analytics.isLoading ? <Spinner /> : (
+          <Fragment>
+            {this.renderCards('Total No. of Travel Requests', {stats: totalRequests, icon: flightIcon} )}
+            {this.renderCards('Total Number of Pending Requests', {stats: pendingRequests, icon: pencdingIcon})}
```
[the remark this answers is not in the document]
Comment (DennisWanjiru) [reply src/views/Analytics/index.jsx]: I have made this change. Thanks for the heads up

File: src/components/AnalyticsCard/__tests__/AnalyticsCard.test.js [hunk 4]
```diff
@@ -15,6 +15,10 @@ describe('<Dashboard />', () => {
   });
 
   it('renders correctly', () => {
-    expect(wrapper).toMatchSnapshot();
+    // expect(wrapper).toMatchSnapshot();
```
[the remark this answers is not in the document]
Comment (DennisWanjiru) [reply src/components/AnalyticsCard/__tests__/AnalyticsCard.test.js]: I have corrected this line of code. Thanks for the highlight

File: src/views/Analytics/index.jsx [hunk 5]
```diff
@@ -10,49 +10,60 @@ import flightLand from '../../images/icons/flight_land.svg';
...
     const { start, end } = context.state.range;
-    const query =`?dateFrom=${start}&dateTo=${end}`;
+    const query = `?location=${context.state.city}&dateFrom=${start}&dateTo=${end}`;
     fetchAnalytics(query);
   }
 
   componentWillReceiveProps(nextProps) {
     const {context, fetchAnalytics} = this.props;
-    if(nextProps.context.state.filter !== context.state.filter) {
+    if(nextProps.context.state.range.start !== context.state.range.start || nextProps.context.state.range.end !== context.state.range.end) {
```
[the remark this answers is not in the document]
Comment (DennisWanjiru) [reply src/views/Analytics/index.jsx]: I have made this change too. Thanks for pointing out that.

**Comment by rovilay** [comment]
Please fix merge conflicts.

**Comment by DennisWanjiru** [comment]
@ikenjoku It is derived from localStorage which is derived from the Andelan API

File: src/components/Spinner/index.scss [hunk 6]
```diff
@@ -0,0 +1,9 @@
+.spinner--large {
+  border: 2px solid #f3f3f3;
+  border-top: 2px solid #385BD3;
+  border-radius: 50%;
+  width: 60px;
+  height: 60px;
+  animation: spin 2s linear infinite;
+  margin-top: 40px;
+}
```
[the remark this answers is not in the document]
Comment (DennisWanjiru) [reply src/components/Spinner/index.scss]: okay on it

File: src/components/CalendarRange/index.jsx [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+      />
+    );
+  }
+}
+
+CalendarRange.propTypes = {
+  handleChange: PropTypes.func.isRequired
+};
+
+// const wrapper = mount(
```
Comment (amarachukwu-agbo) [inline src/components/CalendarRange/index.jsx]: Please remove unused code

File: src/redux/middleware/analyticsSaga.js [hunk 8]
```diff
@@ -23,3 +20,18 @@ export function* fetchAnalyticsSaga({query}) {
 export function* watchFetchAnalytics() {
   yield takeLatest(FETCH_ANALYTICS, fetchAnalyticsSaga);
 }
+
+export function* downloadAnalyticsSaga({query}) {
+  try {
+    const {data} = yield call(AnalyticsAPI.exportAnalytics, query);
+    yield FileSaver.saveAs(data, 'Requests/Trips Analytics');
```
Comment (amarachukwu-agbo) [inline src/redux/middleware/analyticsSaga.js]: Please include toast for download success

File: src/views/Analytics/index.scss [hunk 9]
```diff
@@ -2,7 +2,6 @@
   margin-top: 85px;
   display: -ms-grid;
   display: grid;
-  // grid-template-columns: auto auto auto;
   grid-template-columns: repeat(3, 1fr);
   grid-gap: 30px;
 }
```
Comment (amarachukwu-agbo) [inline src/views/Analytics/index.scss]: Please add a line to the end of the file

File: src/views/Dashboard/DashboardContext/__tests__/FilterContext.test.js [hunk 10]
```diff
@@ -4,45 +4,16 @@ import moment from 'moment';
 import FilterContext from '../FilterContext';
 
 describe('<FilterContext />', () => {
+  localStorage.setItem('location', 'Nairobi, Kenya');
   const wrapper = shallow(<FilterContext />);
+  localStorage.getItem = jest.fn();
```
Comment (amarachukwu-agbo) [inline src/views/Dashboard/DashboardContext/__tests__/FilterContext.test.js]: We have a config file where localStorage is mocked already. I think we can do without this line

File: src/components/CalendarRange/__tests__/CalendarRange.test.js [hunk 11]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { mount } from 'enzyme';
+import sinon from 'sinon';
```
Comment (amarachukwu-agbo) [inline src/components/CalendarRange/__tests__/CalendarRange.test.js]: Please remove 'sinon' since you are using jest for mocks and spys

**Comment by samuelkubai** [comment]
@DennisWanjiru please resolve the merge conflicts.

**Comment by DennisWanjiru** [comment]
@samuelkubai I have resolved the conflicts

**Comment by samuelkubai** [comment]
@amarachukwu-agbo is this good to go?

File: src/components/AnalyticsCard/__tests__/AnalyticsCard.test.js [hunk 12]
```diff
@@ -1,11 +1,12 @@
-import React from 'react';
+import React, { Children } from 'react';
```
Comment (amarachukwu-agbo) [inline src/components/AnalyticsCard/__tests__/AnalyticsCard.test.js]: @DennisWanjiru , please remove the Children you imported as it is not being used

File: src/components/DashboardHeader/index.jsx [hunk 13]
```diff
@@ -1,46 +1,80 @@
...
-          <TimelineDropdown
-            icon={activeCalendar}
-            dropDownItems={['Today', 'Tomorrow','This Week', 'This Month', 'Pick a date']}
-          />
+          {this.renderButton(activeLocation, context.state.city)}
+          {this.renderButton(activeCalendar, `${range.start} - ${range.end}`, this.showCalendar)}
           {this.renderButton(download)}
         </div>
+
+        <div className={`calender-range ${isDropdownOpen ? 'open' : ''}`}>
```
Comment (amarachukwu-agbo) [inline src/components/DashboardHeader/index.jsx]: Why is this code commented out?

File: src/views/Dashboard/index.scss [hunk 14]
```diff
@@ -0,0 +1,3 @@
+html, body, * {
+  font-family: 'Din Pro'!important;
+}
```
Comment (amarachukwu-agbo) [inline src/views/Dashboard/index.scss]: Add a new line

**Comment by amarachukwu-agbo** [comment]
@samuelkubai , LGTM

**Pull request merged**